flcore/datasets.py

```diff
@@ -540,19 +545,158 @@ def load_libsvm(config, center_id=None, task_type="BINARY"):
     test_max_acc = test_unique[1][0] / len(y_test)
     # print(train_max_acc)
     # print(test_max_acc)
-
+    print(X_train.shape, y_train.shape, X_test.shape, y_test.shape)
+    #(271617, 8) (271617,) (157413, 8) (157413,)
+    print(type(X_train), type(y_train), type(X_test),type(y_test))
+    #<class 'numpy.ndarray'> <class 'numpy.ndarray'> <class 'numpy.ndarray'> <class 'numpy.ndarray'>
     return (X_train, y_train), (X_test, y_test)
 
-def load_custom(config):
-    data_file = config["data_file"]
+def std_normalize(col, mean, std):
+    return (col - mean) / std
+
+def iqr_normalize(col, Q1, Q2, Q3):
+    return (col - Q2) / (Q3 - Q1)
+
+def min_max_normalize(col, min_val, max_val):
+    return (col - min_val) / (max_val - min_val)
+
+def load_custom(config,id):
+    with open(config["data_path"]+config['metadata_file'], 'r') as file:
+        metadata = json.load(file)
+
+    data_file = config["data_path"] + config["data_file"]
     ext = data_file.split(".")[-1]
     if ext == "pqt" or ext == "parquet":
         dat = pd.read_parquet(data_file)
     elif ext == "csv":
         dat = pd.read_csv(data_file)
+
     dat_len = len(dat)
 
-    # Print statistics
+<<<<<<< HEAD
+=======
+    # ===================================================
+>>>>>>> 8b3461940f5201da382d50517f44033e013588c3
```

Comment:
Those lines should not be commited, amd will crash the code. Looks like they were left during solving the conflicts between branches.